src/controllers/auth.controller.js

```diff
@@ -0,0 +1,60 @@
+const User = require("../models/User");
+const {
+  generateAccessToken,
+  generateRefreshToken,
+} = require("../utils/jwtUtils");
+
+const { ONE_HOUR_IN_MILLISECONDS } = require("../constants/jwtConstants");
+
+exports.signIn = async function (req, res, next) {
+  const { email, photoURL, displayName } = req.body;
+
+  try {
+    let user = await User.findOne({ email });
+
+    if (!user) {
+      user = new User({
+        email,
+        photoURL,
+        displayName,
+        isAdmin: false,
+      });
+    }
+
+    const accessToken = generateAccessToken(user);
+    const refreshToken = generateRefreshToken(user);
+
+    user.refreshToken = refreshToken;
+
+    await user.save();
```

Comment:
혹시 이부분은 User.create()로 바꿀 수 있는 방법이 있을까요~? 없으면 이대로 진행해도 괜찮을 것 같습니다!

Reply:
```suggestion
    let user = await User.findOne({ email });

    if (!user) {
      user = await User.create({
        email,
        photoURL,
        displayName,
        isAdmin: false,
        refreshToken: generateRefreshToken(user);
      });
    } else {
      const refreshToken = generateRefreshToken(user);
      user.refreshToken = refreshToken;
      await user.save();
    }

    const accessToken = generateAccessToken(user);
   
```
create를 사용하면 위에 처럼 고칠 수 있을 것 같습니다! create를 사용해도 재로그인시 refreshToken을 새로 발급해야 해서 save()를 사용해야 하므로 if else를 안쓰는 이전의 식이 더 괜찮아 보이는데 서호님은 어떡해 생각하시나요?

Reply:
네네! 저도 원래 작업하신 방향대로 진행하시는게 좋을 것 같습니다!